compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2112,6 +2112,12 @@ impl<'tcx> Place<'tcx> {
         self.projection.iter().any(|elem| elem.is_indirect())
     }
 
+    /// If MirPhase >= Derefered and if projection contains Deref,
+    /// It's guaranteed to be in the first place
+    pub fn has_deref(&self) -> bool {
+        self.projection.first() == Some(&PlaceElem::Deref)
```

Comment:
please add a debug assert that none of the other projections are a deref. Just to prevent that someone is accidentally using it from the wrong mir phase. Same for the other `has_deref` function

Reply:
This is still unaddressed

Reply:
Ah sorry about that